clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -1360,6 +1367,9 @@ public void commitSync(Map<TopicPartition, OffsetAndMetadata> offsets, Duration
             Timer requestTimer = time.timer(timeout.toMillis());
             SyncCommitEvent syncCommitEvent = new SyncCommitEvent(offsets, requestTimer);
             CompletableFuture<Void> commitFuture = commit(syncCommitEvent);
+
+            invokePendingAsyncCommits(requestTimer, false);
```

Comment:
In principle, this is only needed in the case where offsets are empty, because otherwise the commitSync itself should make sure that all previous asynccommits are completed, since the commit requests are ordered and going to the same connection.